.github/workflows/protocol-devchain-anvil.yml

```diff
@@ -124,17 +123,31 @@ jobs:
       - name: 'Install packages'
         shell: bash
         run: yarn
-
-      - name: Generate migrations and run devchain
-        if: success() || failure()
-        run: ./scripts/foundry/create_and_migrate_anvil_devchain.sh
-
-      - name: Run integration tests against local anvil devchain
+      
+      # Starting L1 from scratch instead of JSON state to circumvent this Anvil bug https://github.com/foundry-rs/foundry/issues/7502
+      # Install `lsof` dependency, because it's not readily available on CI, but is required by 
+      # `create_and_migrate_anvil_l2_devchain.sh`, because it uses `stop_anvil.sh` to kill 
+      # existing anvil servers.
+      - name: Run integration tests against L1 devchain
         if: success() || failure()
         run: |
+          sudo apt-get update
+          sudo apt-get install -y lsof
+
+          source ./scripts/foundry/constants.sh
+
+          echo "Starting L1 from scratch to circumvent Anvil bug"
+          source ./scripts/foundry/create_and_migrate_anvil_devchain.sh
+
           forge test -vvv \
           --match-path "test-sol/integration/*" \
-          --fork-url http://127.0.0.1:${{ env.ANVIL_PORT }}
+          --fork-url $ANVIL_RPC_URL
+
+          ./scripts/foundry/stop_anvil.sh
+          
+      - name: Generate L1 and L2 migrations
+        if: success() || failure()
+        run: ./scripts/foundry/create_and_migrate_anvil_l2_devchain.sh
```

Comment:
Might be good to create an issue to track adding integration test here. 

I briefly looked at the error that happens when running the integration test on L2, and its because `RevokeCeloAfterL2TransitionTest` assumes that the chain starts on L1 than transitions to L2 within the test. 

This causes the validator contract to fail during initialization since devchain L2 is already active.

Reply:
Great point, that's done here: 
- https://github.com/celo-org/celo-monorepo/issues/11138